src/main/java/duke/TaskAssigner.java

```diff
@@ -45,33 +45,38 @@ public Task assignToDo(String command) {
     public Task assignDeadline(String command) throws DukeException {
         String[] splitTiming = command.split("/by ");
         if (splitTiming.length != 2) {
-            throw new DukeException("Improper Deadline Format! deadline {desc} /by {date}\n");
+            throw new DukeException("Improper Deadline Format! deadline {desc} /by yyyy-mm-dd hhmm\n");
         }
 
         try {
             int d_index = command.indexOf("/by ") + 4;
             String deadline = command.substring(d_index);
+            System.out.println(deadline);
+            String updated_deadline = TimeChecker.updateTime(deadline);
             String d_desc = command.substring(9, d_index - 4);
-            return new Deadline(d_desc, deadline);
+            return new Deadline(d_desc, updated_deadline);
```

Comment:
Might want to consider changing `updated_deadline` to `updatedDeadline` to adhere to the Java coding standard of naming variables in camelCase. I noticed that you used snake_case instead of camelCase in several other areas of your code as well. Might I suggest using checkStyle to detect such cases which could save you a lot of time instead of finding them manually.